index.html

```diff
@@ -31,7 +31,7 @@ <h1 class="display-3">Task Board</h1>
         <form class="modal-content" id="add-task-form">
           <div class="modal-header">
             <h5 class="modal-title" id="exampleModalLabel">New Task</h5>
-            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+            <button type="button" class="close" data-bs-dismiss="modal" aria-label="Close">
```

Comment:
In order to get the close button to dismiss the modal, I made this little change to the data attribute.  You can see the documentation here:  https://getbootstrap.com/docs/5.1/components/modal/#examples